src/test/java/htsjdk/samtools/util/IOUtilTest.java

```diff
@@ -520,4 +542,179 @@ public void testIsBlockCompressedOnJimfs(Path file, boolean checkExtension, bool
+        };
+    }
+
+    @Test(dataProvider = "regExpTests")
+    public void testRegExp(final String regexp, final String[] expected) throws IOException {
+        final String[] allNames = {"5newline5.txt", "Level2.fofn", "example.bam", "ipsum.txt.bgz", "ipsum.txt.bgzipped_with_gzextension.gz", "slurptest.txt", "Level1.fofn", "empty.txt", "ipsum.txt", "ipsum.txt.bgz.wrongextension", "ipsum.txt.gz"};
+        final Path regExpDir = Files.createTempDirectory("regExpDir");
+        regExpDir.toFile().deleteOnExit();
+        final List<String> listExpected = Arrays.asList(expected);
+        final List<File> expectedFiles = new ArrayList<File>();
+        for (String name : allNames) {
+            final Path file = regExpDir.resolve(name);
+            file.toFile().deleteOnExit();
+            file.toFile().createNewFile();
+            if (listExpected.contains(name)) {
+                expectedFiles.add(file.toFile());
+            }
+        }
+        final File[] result = IOUtil.getFilesMatchingRegexp(regExpDir.toFile(), regexp);
+        Assert.assertEqualsNoOrder(result, expectedFiles.toArray());
+    }
+
+    @Test()
+    public void testReadLines() throws IOException {
+        final Path file = Files.createTempFile("tmp", ".txt");
+        file.toFile().deleteOnExit();
+        final int seed = 12394738;
+        final Random rand = new Random(seed);
+        final int nLines = 5;
+        final List<String> lines = new ArrayList<String>();
+        try (final BufferedWriter writer = Files.newBufferedWriter(file)) {
+            for (int i = 0; i < nLines; i++) {
+                final String line = TEST_STRING + Integer.toString(rand.nextInt(100000000));
+                lines.add(line);
+                writer.write(line);
+                writer.newLine();
+            }
+        }
+        final List<String> retLines = new ArrayList<String>();
+        IOUtil.readLines(file.toFile()).forEachRemaining(retLines::add);
+        Assert.assertEquals(retLines, lines);
+    }
+
+    @DataProvider
+    public static Object[][] fileSuffixTests() {
+        return new Object[][]{
+                {TEST_DATA_DIR.resolve("ipsum.txt"), ".txt"},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgz"), ".bgz"},
+                {TEST_DATA_DIR, null}
+        };
+    }
+
+    @Test(dataProvider = "fileSuffixTests")
+    public void testSuffixTest(final Path file, final String expected) {
+        final String ret = IOUtil.fileSuffix(file.toFile());
+        Assert.assertEquals(ret, expected);
+    }
+
+    @Test
+    public void testCopyDirectoryTree() throws IOException {
+        final Path copyToDir = Files.createTempDirectory("copyToDir");
+        copyToDir.toFile().deleteOnExit();
+        IOUtil.copyDirectoryTree(TEST_VARIANT_DIR.toFile(), copyToDir.toFile());
+        final List<String> collect = Files.walk(TEST_VARIANT_DIR).filter(f -> !f.equals(TEST_VARIANT_DIR)).map(p -> p.getFileName().toString()).collect(Collectors.toList());

```

Comment:
is toString needed after getFileName?

Reply:
No, you're right, can make this a List<Path> instead of a List<String> and still works.

Reply:
I updated your comment since github sanitizes things that look like html....

Reply:
ha, oops, thanks